### ISSUE TITLE
[feat] config 기반 실험 관리 구조 설계 및 구현

### ISSUE THREAD

**Issue by joonwoo-yoo** [AIBootcamp24th/document-type-classification-cv-2#7]
## 목적

프로젝트 전반에서 발생할 수 있는 하드코딩을 제거하고 모든 실험 설정(모델, 데이터, 학습 파라미터 등)을 config 파일로 관리할 수 있는 구조를 설계합니다.
이를 통해 코드 수정 없이 config 변경만으로 다양한 실험이 가능하도록 만듭니다.

## 왜 필요한가

현재 구조에서 설정값이 코드에 직접 들어가면 다음과 같은 문제가 발생

- 실험마다 코드 수정 필요 → 비효율 증가
- 협업 시 설정 충돌 발생
- 재현성(reproducibility) 확보 어려움
- 코드 가독성 저하

따라서 "코드는 고정, 실험은 config로 제어" 구조로 전환이 필요

## 목표 구조

config 파일을 역할별로 분리하여 관리

    configs/
    ├─ base.yaml
    ├─ data.yaml
    ├─ train.yaml
    ├─ inference.yaml
    └─ model/
       ├─ resnet50.yaml
       └─ efficientnet_b0.yaml

### base.yaml
- 프로젝트 전역 설정
- seed
- 데이터 경로
- output 경로

### data.yaml
- 이미지 크기
- augmentation 설정
- normalization

### train.yaml
- epoch
- batch size
- learning rate
- optimizer / scheduler

### model/*.yaml
- 모델 종류 (resnet, efficientnet 등)
- pretrained 여부
- dropout 등 모델 옵션

### inference.yaml
- 추론 시 batch size
- checkpoint 경로
- TTA 여부

## 코드 구조 (예정)

    src/
    └─ config.py

- 여러 yaml 파일을 로드하고 병합
- 코드에서는 cfg 객체 하나만 사용

예시:

    cfg.train.batch_size
    cfg.model.name
    cfg.data.image_size

## 작업 내용

- [x] configs 디렉토리 생성
- [x] base.yaml 작성
- [x] data.yaml 작성
- [x] train.yaml 작성
- [x] inference.yaml 작성
- [x] model config 파일 분리
- [x] config 로더 (src/config.py) 구현
- [ ] yaml 병합 구조 구현

## 사용 방식 (목표)

    python -m src.train \
      --base configs/base.yaml \
      --data configs/data.yaml \
      --train configs/train.yaml \
      --model configs/model/resnet50.yaml

→ config만 변경해서 실험 가능

## 핵심 원칙

- 모든 하이퍼파라미터는 config로 관리 (하드코딩 금지)
- 모델, optimizer, scheduler, augmentation 전부 config 기반
- 코드 수정 없이 실험 가능하도록 설계
- 팀원이 이해하기 쉬운 구조 유지

## 기대 효과

- 실험 속도 향상
- 코드 유지보수성 개선
- 협업 효율 증가
- 재현 가능한 실험 환경 구축
